src/dfx/src/lib/builders/custom.rs

```diff
@@ -189,8 +189,16 @@ impl CanisterBuilder for CustomBuilder {
 
 fn run_command(args: Vec<String>, vars: &[super::Env<'_>]) -> DfxResult<()> {
     let (command_name, arguments) = args.split_first().unwrap();
-
-    let mut cmd = std::process::Command::new(command_name);
+    let command_path = Path::new(command_name);
+    let mut cmd = if arguments.is_empty()
+        && command_path.components().count() == 1
+        && command_path.exists()
+    {
+        // handle "command.sh" with no ./
+        Command::new(format!("./{command_name}"))
+    } else {
+        Command::new(command_name)
+    };
```

Comment:
In case the command does not really work (e.g. the mentioned `script.sh --arg` or maybe `curl a && curl b`), can you please also add an explanation to the error message (or somewhere else in our docs) explaining a) that this does not work intentionally and b) what to do instead?

I've had people run into this and simply getting hung up on this a few times already.

Reply:
Also see: https://github.com/dfinity/sdk/issues/2199#issuecomment-1125520224